__tests__/app.test.js

```diff
@@ -0,0 +1,52 @@
+const db = require("../db/connection");
+const app = require("../app");
+const seed = require("../db/seeds/seed");
+const request = require("supertest");
+const {
+  categoryData,
+  commentData,
+  reviewData,
+  userData,
+} = require("../db/data/test-data");
+
+beforeEach(() => seed({ categoryData, commentData, reviewData, userData }));
+afterAll(() => db.end());
+
+describe("/api/categories", () => {
+  test("GET: should respond with a 200 status code", () => {
+    return request(app).get("/api/categories").expect(200);
+  });
+  test("GET: should respond with an array", () => {
+    return request(app)
+      .get("/api/categories")
+      .expect(200)
+      .then(({ body }) => {
+        const { categories } = body;
+        expect(categories).toBeInstanceOf(Array);
+      });
```

Comment:
These are redundant tests as you are expecting 200 in the other tests, and you are using a forEach which is an array method, which proves this for us

Reply:
I have refactored this, but not sure if correct - assertions are now all inside the .then but unable to nest new tests within this. Am I on the wrong track here?